src/types/did_url.rs

```diff
@@ -0,0 +1,223 @@
+//! Convenience Wrapper around [`Url`] for DID URIs according to the [DID Spec](https://www.w3.org/TR/did-core/#did-syntax)
```

Comment:
This works well but the `url` crate which is based on the [whatwg standard](https://url.spec.whatwg.org/)

Ideally we find a crate that already implemented the standard DID is a derivative of, or we can just use regex like did-ethr reference implementation. It seems to me that the `url` crate is more flexible than regex, though.